### PR TITLE
Update SAML IdP getting started guide 

### LINKED ISSUE #38821: Alternative service to `samltest.id` for SAML IdP guide
### Applies To

[Teleport as a SAML identity provider guide](https://goteleport.com/docs/access-controls/idps/saml-guide/)

### Details

The guide uses [samltest.id ](http://samltest.id/), a free reference SAML IdP and SP testing service as a reference service provider to test integration. Unfortunately, the service is broken now. It may come up online again but it leaves bad experience for our docs user when following the guide. 

### How will we know this is resolved?

The guide should show example based on an always reproducible SAML service provider service, maybe with an application runnable in a docker container. 

### Related Issues

n.a

### PR DESCRIPTION
**Pull request by flyinghermit**
- updates sample service provider with iamshowcase.com
- updates guide to show both the Web UI and tctl based configuration flow.

closes https://github.com/gravitational/teleport/issues/38821

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

File: docs/pages/access-controls/idps/saml-guide.mdx [hunk 1]
```diff
@@ -18,56 +18,105 @@ authenticate to external services.
   Reference](./saml-reference.mdx) before proceeding.
 - User with permission to create service provider resource. The preset `editor` role has this permission.
 - SAML application (also known as a SAML service provider or SP) for testing. For this guide, we'll be using
-[samltest.id](https://samltest.id/) as a service provider for Teleport's SAML IdP.
+[RSA Simple Test Service Provider](https://sptest.iamshowcase.com/) (iamshowcase app) - a free test service that
+let's us test Teleport SAML IdP. The test service has a protected page, which can be accessed only after a
```
Comment (ptgott) [inline docs/pages/access-controls/idps/saml-guide.mdx:22]: I don't see the term "iamshowcase app" on the site itself. Can we call it "Simple Test Service Provider"?

Comment (ptgott) [inline docs/pages/access-controls/idps/saml-guide.mdx:22]:
```suggestion
lets us test Teleport SAML IdP. The test service has a protected page, which can be accessed only after a
```

File: docs/pages/access-controls/idps/saml-guide.mdx [hunk 2]
```diff
@@ -18,56 +18,105 @@ authenticate to external services.
...
+user is federated to the site with a valid SAML assertion flow.
```
Comment (ptgott) [inline docs/pages/access-controls/idps/saml-guide.mdx:23]: Should "federated" be "authenticated" here?

File: docs/pages/access-controls/idps/saml-guide.mdx (hunk 1 above)
Comment (flyinghermit) [reply docs/pages/access-controls/idps/saml-guide.mdx]: Updated to refer full name "Simple Test Service Provider" in 
https://github.com/gravitational/teleport/pull/42337/commits/597b995df63cabf2329c0023bd36a68fd90c3044.

The SAML metadata of the site refers "iamshowcase" as entity ID, which is the canonical name of the service provider.

File: docs/pages/access-controls/idps/saml-guide.mdx (hunk 2 above)
Comment (flyinghermit) [reply docs/pages/access-controls/idps/saml-guide.mdx:23]: Federated is the correct term here. 

In SAML, once a user is successfully authenticated in the IdP, they are federated into the service provider. In this case, user is federated to the "Simple Test Service Provider".

**Review by ptgott — APPROVED** [review]

**Pull request merged**